megamek/src/megamek/common/Bay.java

```diff
@@ -79,19 +81,42 @@ public Bay(double space, int doors, int bayNumber) {
         totalSpace = space;
         currentSpace = space;
         this.doors = doors;
-        doorsNext = doors;
+        doorsNext = currentdoors;
         this.bayNumber = bayNumber;
     }
-
+    
+    // For tracking damage to the bay itself, not the cargo it carries
+    public int getbayDamaged() {
+    	return damaged;
+    }
+    
+    public void setbayDamaged() {
+    	damaged = 1;
```

Comment:
If the damage can only be 1 or 0, perhaps make it a boolean instead.

Reply:
I did it this way after having some issues with it being a Boolean in the uniteditor. I think I use the 1 to populate checkboxes, which didn't want to work when it was a Boolean value.